scripts/full_check.sh

```diff
@@ -0,0 +1,108 @@
+#! /bin/bash
+
+# Copyright (C) 2024 Intel Corporation
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+# http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+# 
+# SPDX-License-Identifier: Apache-2.0
+
+
+# this is for developer local check only.
+# not used by github action
+
+check_tool() {
+  which $1 &> /dev/null || (echo "$1 not found!" && exit 1)
+}
+
+for arg in "$@"; do
+  case $arg in
+    -f|--format)
+      CHECK_FORMAT=1
+      ;;
+    -t|--tidy)
+      CHECK_TIDY=1
+      ;;
+    -l|--license)
+      CHECK_LICENSE=1
+      ;;
+    -c|--clean)
+      CLEANUP=1
+      ;;
+    *) # Handle other unknown options
+      echo "Unknown option: $1"
+      exit 1
+      ;;
+  esac
```

Comment:
shall we add another flag to go through all above check, like `-a` or `--all`.

Reply:
@Yun-Fly 
Supported in the following commit.